### ISSUE TITLE
test: add unique IDs to TS rule test cases

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3656]
## Overview

Add unique identifiers to every TS rule test case so that each test can be cross-referenced with its Rust counterpart.

## Tasks

- [ ] Add an ID to every `describe` / `test` / `it` block in rule test files
- [ ] IDs combine the rule name and a sequential classifier (any unique format is acceptable)
- [ ] Issue-related tests (regression tests, etc.) must include the issue number in the ID

## Example

```ts
// Before
it('should report duplicate attributes', ...);

// After
it('[attr-duplication-001] should report duplicate attributes', ...);

// Issue regression test
it('[attr-duplication-issue-1234] regression for #1234', ...);
```

## Target Branch

`dev`

## Related

- #3178 (Rust rewrite tracking issue — enables ID alignment between TS and Rust tests)
